packages/magento-store/utils/redirectOrNotFound.ts

```diff
@@ -67,16 +67,17 @@ export async function redirectOrNotFound(
       candidates.add(from.endsWith(suffix) ? from.slice(0, -suffix.length) : `${from}${suffix}`)
     })
 
-    const routePromises = [...candidates].map(
-      async (url) =>
-        (
+    const routePromises = [...candidates].map(async (url) => {
```

Comment:
Easier solution might be: 

```suggestion
    const routePromises = [...candidates].filter(Boolean).map(async (url) => {
```

Reply:
@paales I've added your suggestion 👍